### PR TITLE
feat(components/colorpicker): add focus trap to colorpicker

### PR DESCRIPTION
**Pull request by Blackbaud-PaulCrowder**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2060/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/forms-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/2060/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2060/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/2060/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2060/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2060/playground/)

File: libs/components/core/src/lib/modules/overlay/overlay.component.ts [hunk 1]
```diff
@@ -120,12 +121,16 @@ export class SkyOverlayComponent implements OnInit, OnDestroy {
...
+  #siblingAriaHiddenCache = new Map<Element, string | null>();
+
   readonly #changeDetector = inject(ChangeDetectorRef);
   readonly #context = inject(SkyOverlayContext);
   readonly #coreAdapter = inject(SkyCoreAdapterService);
+  readonly #adapter = inject(SkyOverlayAdapterService);
   readonly #environmentInjector = inject(EnvironmentInjector);
   readonly #idSvc = inject(SkyIdService);
   readonly #router = inject(Router, { optional: true });
+  readonly #elementRef = inject(ElementRef);
```
Comment (Blackbaud-SteveBrush) [inline libs/components/core/src/lib/modules/overlay/overlay.component.ts]: Mind if we alphabetize these additions?

Comment (Blackbaud-PaulCrowder) [reply libs/components/core/src/lib/modules/overlay/overlay.component.ts]: Done.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick [failed](https://github.com/blackbaud/skyux/actions/runs/8102519711)! :x: Please resolve conflicts and create a new PR.

**Pull request merged**